### ISSUE TITLE
Connection resets provoke ~spurious exception logging

### ISSUE THREAD

**Issue by cemerick** [ztellman/aleph#326]
I've always seen this pair of exceptions logged, but some new traffic has made them less ignorable:

```
 [ERROR] 2017-07-27 05:50:22,138 manifold.stream.graph error in message propagation 
java.io.IOException: Connection reset by peer 
	at sun.nio.ch.FileDispatcherImpl.writev0(Native Method) 
	at sun.nio.ch.SocketDispatcher.writev(SocketDispatcher.java:51) 
	at sun.nio.ch.IOUtil.write(IOUtil.java:148) 
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:504) 
	at io.netty.channel.socket.nio.NioSocketChannel.doWrite(NioSocketChannel.java:432) 
	at io.netty.channel.AbstractChannel$AbstractUnsafe.flush0(AbstractChannel.java:856) 
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.forceFlush(AbstractNioChannel.java:368) 
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:618) 
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:559) 
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:476) 
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:438) 
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858) 
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144) 
	at java.lang.Thread.run(Thread.java:745) 
[ERROR] 2017-07-27 05:50:22,139 manifold.stream.graph error in manifold.utils/future-with 
java.io.IOException: Stream closed 
	at java.util.zip.InflaterInputStream.ensureOpen(InflaterInputStream.java:67) 
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:142) 
	at java.io.FilterInputStream.read(FilterInputStream.java:133) 
	at java.nio.channels.Channels$ReadableByteChannelImpl.read(Channels.java:385) 
	at byte_streams$fn__13786.invokeStatic(byte_streams.clj:701) 
	at byte_streams$fn__13786.invoke(byte_streams.clj:676) 
	at byte_streams.protocols$fn__12731$G__12726__12740.invoke(protocols.clj:9) 
	at byte_streams$fn__13606$f__13409__auto____13608$fn__13610.invoke(byte_streams.clj:452) 
	at clojure.lang.LazySeq.sval(LazySeq.java:40) 
	at clojure.lang.LazySeq.seq(LazySeq.java:49) 
	at clojure.lang.RT.seq(RT.java:525) 
	at clojure.core$seq__6165.invokeStatic(core.clj:137) 
	at clojure.core$map$fn__6623.invoke(core.clj:2712) 
	at clojure.lang.LazySeq.sval(LazySeq.java:40) 
	at clojure.lang.LazySeq.seq(LazySeq.java:49) 
	at clojure.lang.RT.seq(RT.java:525) 
	at clojure.core$seq__6165.invokeStatic(core.clj:137) 
	at clojure.core$empty_QMARK_.invokeStatic(core.clj:6081) 
	at clojure.core$empty_QMARK_.invoke(core.clj:6081) 
	at manifold.stream.seq.SeqSource.take(seq.clj:41) 
	at manifold.stream.graph$sync_connect$f__10878__auto____11827.invoke(graph.clj:255) 
	at clojure.lang.AFn.run(AFn.java:22) 
	at io.aleph.dirigiste.Executor$3.run(Executor.java:318) 
	at io.aleph.dirigiste.Executor$Worker$1.run(Executor.java:62) 
	at manifold.executor$thread_factory$reify__10770$f__10771.invoke(executor.clj:44) 
	at clojure.lang.AFn.run(AFn.java:22) 
	at java.lang.Thread.run(Thread.java:745) 
```

I don't think there's any way to identify connection resets except by sniffing exception messages. ☹️ Aside from that unpleasantness, I think it'd be great if client disconnects didn't result in errors being logged.

**Comment by ztellman** [comment]
Yeah, I don't log `IOException` errors in the Netty code, but it looks like they're leaking out into Manifold in some cases.  I'll give some thought as to how to not make these sorts of errors overwhelming.

**Comment by cemerick** [comment]
These errors got irritating enough that I wanted to squelch them pending whatever long-term resolution is crafted here. Sadly, filtering them via my alerting infra wasn't possible (logs-as-streams of lines are _great_). I was going to fork aleph and do whatever was needed, but 5m in I honestly didn't want to start plucking around with something so essential.

A better workaround then, focusing on the actual irritant, the log messages:

```clojure
(defonce original-log clojure.tools.logging/log*)

(defn- log* [logger level ^Throwable throwable message]
  (let [ns (.getName ^org.slf4j.Logger logger)]
    (when-not (and (= ns "manifold.stream.graph")
                   (instance? java.io.IOException throwable)
                   (#{"Connection reset by peer" "Stream closed"} (.getMessage throwable)))
      (original-log logger level throwable message))))

(defonce ^:private fixup-client-disconnection-logging
         (do (.bindRoot ^clojure.lang.Var #'clojure.tools.logging/log* #'log*)
             true))
```

**Comment by ztellman** [comment]
The first exception is a little curious, since it doesn't have any non-Netty code in the stack trace (is it missing a cause, maybe?).  As to the second, I believe this is actually already fixed in `[byte-streams "0.2.4-alpha3"]`, if you want to give it a try.  Sorry for not getting back to this sooner, please feel free to bother me sooner about it in the future.

**Comment by SevereOverfl0w** [comment]
We have updated byte-streams version, the latter exception appears to have stopped. We are seeing the former however.

I suspect the lack of non-netty frames is due to the nature of the exception being async, and being based directly to the function.
